### PR TITLE
Connect Tree Popup

### PR DESCRIPTION
**Pull request by SofieDunt**
## Why

[Monday.com Ticket](https://code4community-team.monday.com/boards/925293934/views/18904377/pulses/1224885844)

Adds tree popup to map so users can easily navigate to tree pages.

## This PR

MapView
- created a [custom popup](https://developers.google.com/maps/documentation/javascript/examples/overlay-popup)
- changed `useEffect` dependencies so the map doesn't reload whenever the popup is added/popup data is changed (or whenever the reservation modal is shown) by
  - making `markersArray` a constant within `useEffect`
  - making the dependencies `mapRef.current` and `treePopupRef.current` rather than the refs themselves. Since the refs update after DOM loads, created useEffects to update `mapElement` and `treePopupElement` as needed
 
TreePopup
- is not visible by default (otherwise would show on map before click event) and may be closed
- capitalizes species names, adjusts appearance if tree does not have a species name or an address
- overflow for long addresses, max-height so popups with little content are smaller

## Screenshots

 
AllData
NoAddress
NoSpecies

## Verification Steps

 
Interacted with map, made sure:
- tree popups appeared on/when site clicked
- did not appear on other clicks/at any other time
- popups linked to appropriate tree page

### CODE REVIEW

File: src/components/mapPageComponents/mapView/index.tsx [hunk 1]
```diff
@@ -109,6 +131,49 @@ const MapView: React.FC<MapViewProps> = ({
           },
         });
 
+        class Popup extends google.maps.OverlayView {
```
Comment (rymaju) [inline src/components/mapPageComponents/mapView/index.tsx:134]: ❓ I'm not sure about putting a class definition in the body of a function. This really deserves its own file I think.

File: src/components/mapPageComponents/mapView/index.tsx [hunk 2]
```diff
@@ -61,6 +63,13 @@ const MapView: React.FC<MapViewProps> = ({
   );
```
Comment (rymaju) [inline src/components/mapPageComponents/mapView/index.tsx:63]: ❗ Wow there are over 400 lines of code in this react component excluding JSX. That is _way_ too much. Almost all of this should be extracted into another file.

**Review by rymaju — APPROVED** [review]
Looks solid, I think the mapView page definitely needs a lot of refactoring though. Currently ALL of the code in the function body will be rerun on any rerender. This looks like a pretty bad performance problem. We should meet to look into solutions sometime.

File: src/components/mapPageComponents/mapView/index.tsx (hunk 1 above)
Comment (SofieDunt) [reply src/components/mapPageComponents/mapView/index.tsx:134]: Yeah, I'll look more into this. The problem with pulling the class out (and most of the code in this hook) is that `google` isn't defined until the DOM loads.

File: src/components/mapPageComponents/mapView/index.tsx (hunk 2 above)
Comment (SofieDunt) [reply src/components/mapPageComponents/mapView/index.tsx:63]: Agreed, I'll open a ticket for cleanup

**Pull request merged**